### DISCUSSION TITLE
RS485 configuration for a Nixie

### DISCUSSION THREAD

**Discussion by CodeSlinger69** [tagyoureit/nodejs-poolController#804] [Nixie] [answered]
I'm not putting this question in my long thread [here](https://github.com/tagyoureit/nodejs-poolController/discussions/801), as that thread is ... long. And going to get longer before I'm done.

I'm stuck on how to configure RS485 communications in Nixie. For now, I only have the yellow/green RS485 wires from my pump connected to the Building Automation Hat, just to keep things simple. Here's how I've configured the RPi, njsPC, and REM. As you can see from the first pic, zero bytes, lots of reconnects. I've tried swapping green/yellow on the hat. I've confirmed that the jumpers on the TX and RX pins are correct. Any idea what I'm overlooking?

![rs485 1](https://github.com/tagyoureit/nodejs-poolController/assets/10177469/40a4c1bd-7569-46b8-ba3e-e1ce6f784cc4)
![rs485 2](https://github.com/tagyoureit/nodejs-poolController/assets/10177469/20144bb3-a5fe-4546-b3b6-621ac266bd0c)
![rs485 3](https://github.com/tagyoureit/nodejs-poolController/assets/10177469/05de7213-b811-429d-839d-f7bc98a1014b)

**Comment by rstrouse** [comment]
Did you add the pump and assign circuits to it?  I do not see any outbound attempts.  The pump will not respond until it is spoken to.

**Reply by CodeSlinger69** [reply]
No, I did not. This was the epiphany that I was missing. My mind was in the Easytouch mode of operation - in this mode, njcPC appears to poll the RS485 bus and automagically add to its configuration all the equipment and devices that it gets a response from. In Nixie mode, its a do it yourself mindset. 

Now that I've added the pump myself, I have RS485 comms working. Thanks for getting me over this mental hump.

**Comment by CodeSlinger69** [comment]
Well, I thought I had this figured out, but I'm stuck again on the RS485 topic, as I am now working on integrating the Intellichlor into Nixie. I manually added the IC20 to njsPC:

![ic20-1](https://github.com/tagyoureit/nodejs-poolController/assets/10177469/b2450f22-353c-49c5-bcc4-46448e0db72a)

The good:
- The Comms tab shows bytes flowing across the bus, so I know the bus is functioning
- dashPanel reports my current Pump speed.

The bad:
- dashPanel does not show any info for the IC20
- messageManager is empty (and always has been, even when just the pump was connected to the RS485 bus)

![ic20-2](https://github.com/tagyoureit/nodejs-poolController/assets/10177469/3f4ea166-eacb-41ba-a291-c2291979f797)

![ic20-3](https://github.com/tagyoureit/nodejs-poolController/assets/10177469/d0e171e6-b6a2-4cbb-a238-2a4aa777a485)

One thought maybe; the IC20 surge board is connected to the RS485 bus including the ground and +15V lines. Does the IC20 require the ground and +15 wires to be connected for it to communicate on the RS485 bus? Right now I have only the A and B lines connected to the bus. If the IC20 requires that 15v line, I'll need to find a source for 15V in the box somewhere.

One other question: Where should I look to get messageManager working?

Thanks for your patience with me on this!

**Reply by rstrouse** [reply, marked as answer]
You can find the message manager instructions in the wiki below.
https://github.com/rstrouse/nodejs-poolController-dashPanel/wiki/Message-Manager

Keep in mind that your IC20 will not start communicating until after it has finished its internal sequences and reported the salt levels.  However, please be sure that when you installed it you told it Nixie was its master.  Chlorinators can be installed as if they belong to another panel.

**Reply by CodeSlinger69** [reply]
Ack.

1. messageManager started working after I started a replay capture in dashPanel. Not sure why, but I can see messages flowing in messageManager now, so all good. Must have been some transient thing.
2. IC20 shows in dashPanel now, thanks. I selected External rather than Nixie, which was the problem. I'll document this in my other thread for others.

Thanks.
